src/CabanaPD_Solver.hpp

```diff
@@ -294,15 +279,21 @@ class SolverElastic
             log( std::cout, step, "/", num_steps, " ", std::scientific,
                  std::setprecision( 2 ), step * dt );
 
-            total_time = total_timer.seconds();
-            double rate = 1.0 * particles->n_global * output_frequency /
-                          ( total_time - last_time );
+            double step_time = _step_timer.time();
+            double comm_time = comm->time();
+            double integrate_time = integrator->time();
+            double force_time = force->time();
+            double energy_time = force->timeEnergy();
+            double output_time = particles->timeOutput();
+            _total_time += step_time;
+            double rate =
+                1.0 * particles->n_global * output_frequency / ( step_time );
```

Comment:
Is the "1.0" to convert integer to double?

Reply:
Should be a cast instead - fixed